### PR TITLE
Add Pa11y accessibility test tool to CircleCI pipeline

### PR DESCRIPTION
**Pull request by Gugandeep**
## Purpose
Add Pa11y Accessibility testing tool to pipeline to alert developers to any areas of the service that may cause issues to users who are using other tools to view webpages.

Fixes DDPB-3325

## Approach
Added a pa11y job into the config for CircleCI. This jobs runs the pa11y container/service. The container grabs all the dependencies needed to run the pa11y tool and brings in the config file pa11y needs. 

After the pa11y container is finished being brought up, prod mode is enabled is remove the symfony profiler from display. The job waits for a small amount of time before running the pa11y test.

During the test, pa11y will report any errors it detects. The command to run the pa11y test ensures that the job, and therefore the pipeline, does not fail when an error is detected.

## Learning
Learnt how to create and configure jobs in CircleCI through the config.yml file. 
Learnt how to create a Docker service and configure dependencies and networks.
Learnt how to install packages in a Docker container through the Dockerfile.
Learnt how to configure pa11y tool to run on multiple URLs.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -486,6 +491,26 @@ jobs:
...
+            ~/project/.circleci/set_env.sh >> $BASH_ENV
+      - run:
+          name: Run pa11y
+          command: |
+            docker-compose -f docker-compose.yml up -d pa11y
+            docker-compose exec frontend touch /var/www/.enableProdMode
+            docker-compose exec admin touch /var/www/.enableProdMode
+            docker-compose exec api touch /var/www/.enableProdMode
+            sleep 10
+            docker-compose -f docker-compose.yml run pa11y pa11y-ci || echo "Pa11y found some errors"
```
Comment (acsauk-mojd) [inline .circleci/config.yml:512]: I think it's worth adding a comment here to say what's going on with the double pipe for our future colleague's sake.

File: pa11y/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Installs latest Chromium (77) package.
+RUN apk add --no-cache \
+      chromium \
+      nss \
+      freetype \
+      freetype-dev \
+      harfbuzz \
+      ca-certificates \
+      ttf-freefont \
+      nodejs \
```
Comment (acsauk-mojd) [inline pa11y/Dockerfile:12]: I *think* we can remove this as we're already pulling node in from the base image - worth testing locally.

File: pa11y/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      yarn
```
Comment (acsauk-mojd) [inline pa11y/Dockerfile:13]: We could *possibly* remove this and just use npm that comes with node - not sure if it changes how the package managing works but could save some time for the circle job to run if we only have to worry about one package manager.

File: .circleci/config.yml [hunk 4]
```diff
@@ -486,6 +491,26 @@ jobs:
...
+          command: |
+            ~/project/.circleci/set_env.sh >> $BASH_ENV
+      - run:
+          name: Run pa11y
+          command: |
+            docker-compose -f docker-compose.yml up -d pa11y
+            docker-compose exec frontend touch /var/www/.enableProdMode
+            docker-compose exec admin touch /var/www/.enableProdMode
+            docker-compose exec api touch /var/www/.enableProdMode
+            sleep 10
```
Comment (mattmachell) [inline .circleci/config.yml:511]: I'd tend to avoid sleep and prefer something that waits for a service to be ready.

File: pa11y/Dockerfile (hunk 2 above)
Comment (mattmachell) [reply pa11y/Dockerfile:12]: Might also be worth looking if there's an official alpine node chromium somewhere (Selenium docker builds might have that for example).

File: .circleci/config.yml (hunk 4 above)
Comment (acsauk-mojd) [reply .circleci/config.yml:511]: Good point actually - we're using `waitforit` in Serve-OPG if you want to see how that would work @Gugandeep https://github.com/ministryofjustice/serve-opg/blob/master/.circleci/config.yml

File: .circleci/config.yml (hunk 1 above)
Comment (Gugandeep) [reply .circleci/config.yml:512]: Added a comment to explain what the purpose of the double pipe is

File: pa11y/Dockerfile (hunk 3 above)
Comment (Gugandeep) [reply pa11y/Dockerfile]: Removed installation of nodejs and yarn packages. Changed yarn command to npm install.

File: pa11y/Dockerfile (hunk 2 above)
Comment (Gugandeep) [reply pa11y/Dockerfile]: I've had a look on Docker but couldn't find any official images that fit the match unfortunately.

File: .circleci/config.yml (hunk 4 above)
Comment (Gugandeep) [reply .circleci/config.yml:511]: Added a ticket to the backlog to implement this: https://opgtransform.atlassian.net/browse/DDPB-3518

**Pull request merged**